### ISSUE TITLE
Numpy FutureWarning when setting bins in ImageItem.getHistogram()

### ISSUE THREAD

**Issue by titusjan** [pyqtgraph/pyqtgraph#835]
If I set the `bins` parameter of the `ImageItem.getHistogram()` method to a Numpy array, I get the following Future warning...

```
/Users/kenter/prog/py/pyqtgraph/pyqtgraph/graphicsItems/ImageItem.py:493: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  if bins == 'auto':
```

The solution seems to me to replace `if bins == 'auto'` with `if bins is 'auto'`. 

I didn't make a pull request since it's only one line, but if you like I can make a PR. Just let me know.

I use Numpy 1.15.4 and PyQtGraph 0.10.

**Comment by flutefreak7** [comment]
While `is` can work for string comparison it can't be relied upon (for example: https://stackoverflow.com/a/1504742/1639671)

I would just add an `isinstance(bins, str)` to avoid a numpy array getting compared to 'auto'. That or `not isinstance(bins, np.ndarray)`, but there are so many kinds of sequences (like pandas series) that might not enjoy being compared to a strip, it makes more sense to just require it be a stranger or str subclass to use the auto option.

**Comment by titusjan** [comment]
I thought that it could be relied on because Python maintains a collection of string constants. But I realize now that you can also get to the same result by constructing strings. Obvious in retrospect. Thanks.

**Comment by titusjan** [comment]
@j9ac9k: again, sorry for not responding earlier but PR 921 doesn't remove the warning. Somehow the Numpy comparison is still used, even if the string is on the left hand side. Can we reopen the issue?

To reproduce the problem type the following on the IPython or Python command prompt.

```
In [1]: import numpy as np

In [2]: np.__version__
Out[2]: '1.15.4'

In [3]: bins = np.arange(5)

In [4]: "auto" == bins
/Users/kenter/miniconda3/bin/ipython:1: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  #!/Users/kenter/miniconda3/bin/python
Out[4]: False
```
I don't know what the most elegant fix is. Perhaps something like:

```
if type(bins) == str and bins == "auto":
    ....
```

This is cumbersome though.

**Comment by j9ac9k** [comment]
Re-opened, thanks for following up

**Comment by 2xB** [comment]
Thank you, I have no idea why I was sure to have it fixed and tested in that PR, when it obviously is not. I also have no idea what the design decision behind this for Python itself is (i.e. why `str.__eq__` is `NotImplemented`, when Python obviously has some type of string equality checking built in) - so I asked why this is and how to bypass it on stackoverflow: https://stackoverflow.com/questions/57549477/how-to-call-the-default-string-equality-function-in-python

**Comment by 2xB** [comment]
Okay, the stackoverflow answer is very reasonable: `str.__eq__(np.ndarray)` is `NotImplemented`, while e.g. `str.__eq__(str)` is. Therefore probably the concept by @titusjan would be the one to use...

**Comment by flutefreak7** [comment]
Agreed, though using isinstance() is preferable to explicitly checking the type. `if isinstance(bins, str) and bins == 'auto':`.

**Comment by j9ac9k** [comment]
Would this not be a good time to use `np.all`?

**Comment by 2xB** [comment]
Well, @flutefreak7 s suggestion is faster than `np.all`, since it does not do the element-wise comparison if `bins` is a NumPy array.

**Comment by j9ac9k** [comment]
I apologize, I hadn't looked too closely at this issue.  Is the underlying issue that `bins` can be either a `str` or a `np.ndarray`?  If so, we should definitely use what @flutefreak7 suggested `if isinstance(bins, str) and bins == 'auto'` then do stuff.

**Comment by 2xB** [comment]
Yes, exactly that is the issue.

**Comment by 2xB** [comment]
PR #1024 suggests @flutefreak7 s code. @titusjan Could you quickly verify that this now really solves the issue?

**Comment by j9ac9k** [comment]
@2xB Thanks for doing the PR, I was going to do it last night but decided to go to sleep instead :)

I think this may be a good use case for adding a test though.

**Comment by 2xB** [comment]
Does the CI currently fail on Python warnings?

**Comment by j9ac9k** [comment]
CI doesn't fail on a warning, but it does report it.  I certainly could change the CI process to fail on a warning (should I do that for #991 ?)

pytest is already configured to ignore a handful of warnings.

**Comment by 2xB** [comment]
Since we are opening a new topic here, I created a new issue for it. Failing on warning is very reasonable in my opinion.

**Comment by titusjan** [comment]
@2xB : PR #1024 fixes the issue. The warning is gone. Thanks.
